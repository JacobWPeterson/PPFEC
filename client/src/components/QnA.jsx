import React from 'react';
import styled from 'styled-components';
import QuestionList from './QnAComponents/QuestionList.jsx';
import MoreQ from './QnAComponents/MoreQ.jsx';
import Search from './QnAComponents/Search.jsx';
import AddQ from './QnAComponents/AddQ.jsx';

const QnAContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 65vw;
  margin: 0 auto;
  color: #424242;
`;

const QnATitle = styled.div`
  padding: 2vh 0;
`;

const BottomWrap = styled.div`
  display: flex;
  flex-direction: row;
`;

class QnA extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: this.props.questions.results,
      dd: [],
      max: 0,
      displayed: 0,
      buttonDisplay: true,
    };
    this.loadTwoItems = this.loadTwoItems.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
    this.resetDefault = this.resetDefault.bind(this);
  }

  componentDidMount() {
    this.loadTwoItems();
  }

  handleSearch(string) {
    this.setState({ dd: [] }, () => {
      const results = [];
      this.state.data.forEach((element, index) => {
        if (element.question_body.includes(string)) {
          results.push(this.state.data[index]);
        }
        let count = 0;
        Object.entries(element.answers).forEach((answer) => {
          if (answer[1].body.includes(string)) {
            const obj = {};
            Object.assign(obj, this.state.data[index]);
            obj.answers = { [answer[0]]: answer[1] };
            count += 1;
            obj.question_id += `a${count}`;
            results.push(obj);
          }
        });
      });
      this.setState({
        dd: results,
        buttonDisplay: false,
      });
    });
  }

  loadTwoItems() {
    this.state.max += 2;
    while (this.state.displayed < this.state.max) {
      if (!this.state.data[this.state.displayed]) {
        break;
      }
      this.state.dd.push(this.state.data[this.state.displayed]);
      this.state.displayed += 1;
    }
    if (!this.state.data[this.state.displayed]) {
      this.state.buttonDisplay = false;
    }
    this.setState((prevState) => ({
      dd: prevState.dd,
      displayed: prevState.displayed,
      max: prevState.max,
      buttonDisplay: prevState.buttonDisplay,
    }));
  }

  resetDefault() {
    this.setState({
      dd: [],
      max: 0,
      displayed: 0,
      buttonDisplay: true,
    }, () => (this.loadTwoItems()));
  }

  render() {
    return (
      <QnAContainer>
        <QnATitle>QUESTIONS & ANSWERS</QnATitle>
        <Search func={this.handleSearch} reset={this.resetDefault} />
        <QuestionList items={this.state.dd} />
        <BottomWrap>
          <MoreQ func={this.loadTwoItems} buttonDisplay={this.state.buttonDisplay} />
          <AddQ />
        </BottomWrap>
      </QnAContainer>
    );
  }
}

export default QnA;
