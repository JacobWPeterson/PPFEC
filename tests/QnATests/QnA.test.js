import React from 'react';
import { mount } from 'enzyme';
import QnA from '../../client/src/components/QnA.jsx';

describe('This is for the Add Questions', () => {
  const props = {
    questions:{
      product_id: "14037",
      results: [
        {
          answers: {
            768676: {
              answerer_name: 'Koby72',
              body: 'Laborum voluptatum aut et ex occaecati magni quas.',
              date: '2020-07-28T00:00:00.000Z',
              helpfulness: 15,
              id: 768676,
              photos: [],
            },
            768699: {
              answerer_name: 'Ko2',
              body: 'Laborum volupt aut et ex occaecati magni quas.',
              date: '2020-07-28T00:00:00.000Z',
              helpfulness: 15,
              id: 768699,
              photos: [],
            },
          },
          asker_name: 'Maxine.Kilback16',
          question_body: 'Fugit in recusandae delectus ut.',
          question_date: '2020-03-19T00:00:00.000Z',
          question_helpfulness: 30,
          question_id: 80975,
          reported: false,
        },
        {
          answers: {
            768676: {
              answerer_name: 'Koby72',
              body: 'Laborum voluptatum aut et ex occaecati magni quas.',
              date: '2020-07-28T00:00:00.000Z',
              helpfulness: 15,
              id: 768676,
              photos: [],
            },
            768699: {
              answerer_name: 'Ko2',
              body: 'Laborum volupt aut et ex occaecati magni quas.',
              date: '2020-07-28T00:00:00.000Z',
              helpfulness: 15,
              id: 768699,
              photos: [],
            },
          },
          asker_name: 'Maxine.Kilback16',
          question_body: 'Fugit in recusandae delectus ut.',
          question_date: '2020-03-19T00:00:00.000Z',
          question_helpfulness: 30,
          question_id: 80945,
          reported: false,
        },
      ],
    },
    name: 'This is a product',
  };

  const QnAComp = mount(<QnA {...props}/>);

  it('Should render Add Question Properly', () => {
    expect(QnAComp).toMatchSnapshot();
  });
});