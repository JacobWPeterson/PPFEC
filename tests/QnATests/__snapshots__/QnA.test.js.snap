// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is for the Add Questions Should render Add Question Properly 1`] = `
<QnA
  name="This is a product"
  questions={
    Object {
      "product_id": "14037",
      "results": Array [
        Object {
          "answers": Object {
            "768676": Object {
              "answerer_name": "Koby72",
              "body": "Laborum voluptatum aut et ex occaecati magni quas.",
              "date": "2020-07-28T00:00:00.000Z",
              "helpfulness": 15,
              "id": 768676,
              "photos": Array [],
            },
            "768699": Object {
              "answerer_name": "Ko2",
              "body": "Laborum volupt aut et ex occaecati magni quas.",
              "date": "2020-07-28T00:00:00.000Z",
              "helpfulness": 15,
              "id": 768699,
              "photos": Array [],
            },
          },
          "asker_name": "Maxine.Kilback16",
          "question_body": "Fugit in recusandae delectus ut.",
          "question_date": "2020-03-19T00:00:00.000Z",
          "question_helpfulness": 30,
          "question_id": 80975,
          "reported": false,
        },
        Object {
          "answers": Object {
            "768676": Object {
              "answerer_name": "Koby72",
              "body": "Laborum voluptatum aut et ex occaecati magni quas.",
              "date": "2020-07-28T00:00:00.000Z",
              "helpfulness": 15,
              "id": 768676,
              "photos": Array [],
            },
            "768699": Object {
              "answerer_name": "Ko2",
              "body": "Laborum volupt aut et ex occaecati magni quas.",
              "date": "2020-07-28T00:00:00.000Z",
              "helpfulness": 15,
              "id": 768699,
              "photos": Array [],
            },
          },
          "asker_name": "Maxine.Kilback16",
          "question_body": "Fugit in recusandae delectus ut.",
          "question_date": "2020-03-19T00:00:00.000Z",
          "question_helpfulness": 30,
          "question_id": 80945,
          "reported": false,
        },
      ],
    }
  }
>
  <QnA__QnAContainer>
    <div
      className="QnA__QnAContainer-sc-1hcdygh-0 bbNvMT"
    >
      <QnA__QnATitle>
        <div
          className="QnA__QnATitle-sc-1hcdygh-1 dZBFqZ"
        >
          QUESTIONS & ANSWERS
        </div>
      </QnA__QnATitle>
      <Search
        func={[Function]}
        reset={[Function]}
      >
        <Search__Wrapper>
          <label
            className="Search__Wrapper-np7a72-1 biulzn"
          >
            <Search__Input
              onChange={[Function]}
              placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
              type="text"
              value=""
            >
              <input
                className="Search__Input-np7a72-0 iZcJlO"
                onChange={[Function]}
                placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
                type="text"
                value=""
              />
            </Search__Input>
            <Search__Magnify>
              <div
                className="Search__Magnify-np7a72-2 jCFwuN"
              >
                âš²
              </div>
            </Search__Magnify>
          </label>
        </Search__Wrapper>
      </Search>
      <QuestionList
        items={
          Array [
            Object {
              "answers": Object {
                "768676": Object {
                  "answerer_name": "Koby72",
                  "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                  "date": "2020-07-28T00:00:00.000Z",
                  "helpfulness": 15,
                  "id": 768676,
                  "photos": Array [],
                },
                "768699": Object {
                  "answerer_name": "Ko2",
                  "body": "Laborum volupt aut et ex occaecati magni quas.",
                  "date": "2020-07-28T00:00:00.000Z",
                  "helpfulness": 15,
                  "id": 768699,
                  "photos": Array [],
                },
              },
              "asker_name": "Maxine.Kilback16",
              "question_body": "Fugit in recusandae delectus ut.",
              "question_date": "2020-03-19T00:00:00.000Z",
              "question_helpfulness": 30,
              "question_id": 80975,
              "reported": false,
            },
            Object {
              "answers": Object {
                "768676": Object {
                  "answerer_name": "Koby72",
                  "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                  "date": "2020-07-28T00:00:00.000Z",
                  "helpfulness": 15,
                  "id": 768676,
                  "photos": Array [],
                },
                "768699": Object {
                  "answerer_name": "Ko2",
                  "body": "Laborum volupt aut et ex occaecati magni quas.",
                  "date": "2020-07-28T00:00:00.000Z",
                  "helpfulness": 15,
                  "id": 768699,
                  "photos": Array [],
                },
              },
              "asker_name": "Maxine.Kilback16",
              "question_body": "Fugit in recusandae delectus ut.",
              "question_date": "2020-03-19T00:00:00.000Z",
              "question_helpfulness": 30,
              "question_id": 80945,
              "reported": false,
            },
          ]
        }
        name="This is a product"
      >
        <QuestionList__QList>
          <div
            className="QuestionList__QList-ar56kx-0 eOppjS"
          >
            <Item
              id={80975}
              item={
                Object {
                  "answers": Object {
                    "768676": Object {
                      "answerer_name": "Koby72",
                      "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                      "date": "2020-07-28T00:00:00.000Z",
                      "helpfulness": 15,
                      "id": 768676,
                      "photos": Array [],
                    },
                    "768699": Object {
                      "answerer_name": "Ko2",
                      "body": "Laborum volupt aut et ex occaecati magni quas.",
                      "date": "2020-07-28T00:00:00.000Z",
                      "helpfulness": 15,
                      "id": 768699,
                      "photos": Array [],
                    },
                  },
                  "asker_name": "Maxine.Kilback16",
                  "question_body": "Fugit in recusandae delectus ut.",
                  "question_date": "2020-03-19T00:00:00.000Z",
                  "question_helpfulness": 30,
                  "question_id": 80975,
                  "reported": false,
                }
              }
              key="80975"
              name="This is a product"
            >
              <div>
                <Item__TopWrapper>
                  <div
                    className="Item__TopWrapper-nd8d2l-0 dafvsE"
                  >
                    <Item__QB>
                      <div
                        className="Item__QB-nd8d2l-8 fxalhT"
                      >
                        <Item__Q>
                          <div
                            className="Item__Q-nd8d2l-7 xnwUm"
                          >
                            Q:
                          </div>
                        </Item__Q>
                        Fugit in recusandae delectus ut.
                      </div>
                    </Item__QB>
                    <Item__Helper>
                      <div
                        className="Item__Helper-nd8d2l-2 cFkmUj"
                      >
                        <div>
                          Helpful?
                        </div>
                        <Item__Yes
                          onClick={[Function]}
                        >
                          <div
                            className="Item__Yes-nd8d2l-3 czRgbS"
                            onClick={[Function]}
                          >
                            Yes
                          </div>
                        </Item__Yes>
                        <Item__Helpfulness>
                          <div
                            className="Item__Helpfulness-nd8d2l-5 jZuxNN"
                          >
                            (30)
                          </div>
                        </Item__Helpfulness>
                        <div>
                          |
                        </div>
                        <Item__AddAButton
                          onClick={[Function]}
                        >
                          <div
                            className="Item__AddAButton-nd8d2l-4 jOspkh"
                            onClick={[Function]}
                          >
                            Add Answer
                          </div>
                        </Item__AddAButton>
                      </div>
                    </Item__Helper>
                  </div>
                </Item__TopWrapper>
                <Item__BottomWrapper>
                  <div
                    className="Item__BottomWrapper-nd8d2l-1 iZhiUt"
                  >
                    <Item__A>
                      <div
                        className="Item__A-nd8d2l-6 kwDvjH"
                      >
                        A:
                      </div>
                    </Item__A>
                    <Item__ListContainer>
                      <div
                        className="Item__ListContainer-nd8d2l-9 GlEbY"
                      >
                        <AnswerList
                          answers={
                            Array [
                              Object {
                                "answerer_name": "Koby72",
                                "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                                "date": "2020-07-28T00:00:00.000Z",
                                "helpfulness": 15,
                                "id": 768676,
                                "photos": Array [],
                              },
                              Object {
                                "answerer_name": "Ko2",
                                "body": "Laborum volupt aut et ex occaecati magni quas.",
                                "date": "2020-07-28T00:00:00.000Z",
                                "helpfulness": 15,
                                "id": 768699,
                                "photos": Array [],
                              },
                            ]
                          }
                        >
                          <div>
                            <Answer
                              answer={
                                Object {
                                  "answerer_name": "Koby72",
                                  "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                                  "date": "2020-07-28T00:00:00.000Z",
                                  "helpfulness": 15,
                                  "id": 768676,
                                  "photos": Array [],
                                }
                              }
                              id={768676}
                              key="768676"
                            >
                              <div>
                                <Answer__Body>
                                  <div
                                    className="Answer__Body-sc-3mvxjy-6 gnmbP"
                                  >
                                    Laborum voluptatum aut et ex occaecati magni quas.
                                  </div>
                                </Answer__Body>
                                <div>
                                  <Photos
                                    photos={Array []}
                                  >
                                    <div />
                                  </Photos>
                                </div>
                                <Answer__BottomContainer>
                                  <div
                                    className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                                  >
                                    <Answer__NameDate>
                                      <div
                                        className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                      >
                                        By Koby72 on  July, 28 2020
                                      </div>
                                    </Answer__NameDate>
                                    |
                                    <Answer__Helpful>
                                      <div
                                        className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                      >
                                        Helpful?
                                      </div>
                                    </Answer__Helpful>
                                    <Answer__Yes
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                        onClick={[Function]}
                                      >
                                        Yes
                                      </div>
                                    </Answer__Yes>
                                    <Answer__Help>
                                      <div
                                        className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                      >
                                        (15)
                                      </div>
                                    </Answer__Help>
                                    |
                                    <Answer__Report
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                        onClick={[Function]}
                                      >
                                        Report
                                      </div>
                                    </Answer__Report>
                                  </div>
                                </Answer__BottomContainer>
                              </div>
                            </Answer>
                            <Answer
                              answer={
                                Object {
                                  "answerer_name": "Ko2",
                                  "body": "Laborum volupt aut et ex occaecati magni quas.",
                                  "date": "2020-07-28T00:00:00.000Z",
                                  "helpfulness": 15,
                                  "id": 768699,
                                  "photos": Array [],
                                }
                              }
                              id={768699}
                              key="768699"
                            >
                              <div>
                                <Answer__Body>
                                  <div
                                    className="Answer__Body-sc-3mvxjy-6 gnmbP"
                                  >
                                    Laborum volupt aut et ex occaecati magni quas.
                                  </div>
                                </Answer__Body>
                                <div>
                                  <Photos
                                    photos={Array []}
                                  >
                                    <div />
                                  </Photos>
                                </div>
                                <Answer__BottomContainer>
                                  <div
                                    className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                                  >
                                    <Answer__NameDate>
                                      <div
                                        className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                      >
                                        By Ko2 on  July, 28 2020
                                      </div>
                                    </Answer__NameDate>
                                    |
                                    <Answer__Helpful>
                                      <div
                                        className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                      >
                                        Helpful?
                                      </div>
                                    </Answer__Helpful>
                                    <Answer__Yes
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                        onClick={[Function]}
                                      >
                                        Yes
                                      </div>
                                    </Answer__Yes>
                                    <Answer__Help>
                                      <div
                                        className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                      >
                                        (15)
                                      </div>
                                    </Answer__Help>
                                    |
                                    <Answer__Report
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                        onClick={[Function]}
                                      >
                                        Report
                                      </div>
                                    </Answer__Report>
                                  </div>
                                </Answer__BottomContainer>
                              </div>
                            </Answer>
                          </div>
                        </AnswerList>
                        <MoreA
                          display={false}
                          func={[Function]}
                        >
                          <div />
                        </MoreA>
                      </div>
                    </Item__ListContainer>
                  </div>
                </Item__BottomWrapper>
              </div>
            </Item>
            <Item
              id={80945}
              item={
                Object {
                  "answers": Object {
                    "768676": Object {
                      "answerer_name": "Koby72",
                      "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                      "date": "2020-07-28T00:00:00.000Z",
                      "helpfulness": 15,
                      "id": 768676,
                      "photos": Array [],
                    },
                    "768699": Object {
                      "answerer_name": "Ko2",
                      "body": "Laborum volupt aut et ex occaecati magni quas.",
                      "date": "2020-07-28T00:00:00.000Z",
                      "helpfulness": 15,
                      "id": 768699,
                      "photos": Array [],
                    },
                  },
                  "asker_name": "Maxine.Kilback16",
                  "question_body": "Fugit in recusandae delectus ut.",
                  "question_date": "2020-03-19T00:00:00.000Z",
                  "question_helpfulness": 30,
                  "question_id": 80945,
                  "reported": false,
                }
              }
              key="80945"
              name="This is a product"
            >
              <div>
                <Item__TopWrapper>
                  <div
                    className="Item__TopWrapper-nd8d2l-0 dafvsE"
                  >
                    <Item__QB>
                      <div
                        className="Item__QB-nd8d2l-8 fxalhT"
                      >
                        <Item__Q>
                          <div
                            className="Item__Q-nd8d2l-7 xnwUm"
                          >
                            Q:
                          </div>
                        </Item__Q>
                        Fugit in recusandae delectus ut.
                      </div>
                    </Item__QB>
                    <Item__Helper>
                      <div
                        className="Item__Helper-nd8d2l-2 cFkmUj"
                      >
                        <div>
                          Helpful?
                        </div>
                        <Item__Yes
                          onClick={[Function]}
                        >
                          <div
                            className="Item__Yes-nd8d2l-3 czRgbS"
                            onClick={[Function]}
                          >
                            Yes
                          </div>
                        </Item__Yes>
                        <Item__Helpfulness>
                          <div
                            className="Item__Helpfulness-nd8d2l-5 jZuxNN"
                          >
                            (30)
                          </div>
                        </Item__Helpfulness>
                        <div>
                          |
                        </div>
                        <Item__AddAButton
                          onClick={[Function]}
                        >
                          <div
                            className="Item__AddAButton-nd8d2l-4 jOspkh"
                            onClick={[Function]}
                          >
                            Add Answer
                          </div>
                        </Item__AddAButton>
                      </div>
                    </Item__Helper>
                  </div>
                </Item__TopWrapper>
                <Item__BottomWrapper>
                  <div
                    className="Item__BottomWrapper-nd8d2l-1 iZhiUt"
                  >
                    <Item__A>
                      <div
                        className="Item__A-nd8d2l-6 kwDvjH"
                      >
                        A:
                      </div>
                    </Item__A>
                    <Item__ListContainer>
                      <div
                        className="Item__ListContainer-nd8d2l-9 GlEbY"
                      >
                        <AnswerList
                          answers={
                            Array [
                              Object {
                                "answerer_name": "Koby72",
                                "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                                "date": "2020-07-28T00:00:00.000Z",
                                "helpfulness": 15,
                                "id": 768676,
                                "photos": Array [],
                              },
                              Object {
                                "answerer_name": "Ko2",
                                "body": "Laborum volupt aut et ex occaecati magni quas.",
                                "date": "2020-07-28T00:00:00.000Z",
                                "helpfulness": 15,
                                "id": 768699,
                                "photos": Array [],
                              },
                            ]
                          }
                        >
                          <div>
                            <Answer
                              answer={
                                Object {
                                  "answerer_name": "Koby72",
                                  "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                                  "date": "2020-07-28T00:00:00.000Z",
                                  "helpfulness": 15,
                                  "id": 768676,
                                  "photos": Array [],
                                }
                              }
                              id={768676}
                              key="768676"
                            >
                              <div>
                                <Answer__Body>
                                  <div
                                    className="Answer__Body-sc-3mvxjy-6 gnmbP"
                                  >
                                    Laborum voluptatum aut et ex occaecati magni quas.
                                  </div>
                                </Answer__Body>
                                <div>
                                  <Photos
                                    photos={Array []}
                                  >
                                    <div />
                                  </Photos>
                                </div>
                                <Answer__BottomContainer>
                                  <div
                                    className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                                  >
                                    <Answer__NameDate>
                                      <div
                                        className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                      >
                                        By Koby72 on  July, 28 2020
                                      </div>
                                    </Answer__NameDate>
                                    |
                                    <Answer__Helpful>
                                      <div
                                        className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                      >
                                        Helpful?
                                      </div>
                                    </Answer__Helpful>
                                    <Answer__Yes
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                        onClick={[Function]}
                                      >
                                        Yes
                                      </div>
                                    </Answer__Yes>
                                    <Answer__Help>
                                      <div
                                        className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                      >
                                        (15)
                                      </div>
                                    </Answer__Help>
                                    |
                                    <Answer__Report
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                        onClick={[Function]}
                                      >
                                        Report
                                      </div>
                                    </Answer__Report>
                                  </div>
                                </Answer__BottomContainer>
                              </div>
                            </Answer>
                            <Answer
                              answer={
                                Object {
                                  "answerer_name": "Ko2",
                                  "body": "Laborum volupt aut et ex occaecati magni quas.",
                                  "date": "2020-07-28T00:00:00.000Z",
                                  "helpfulness": 15,
                                  "id": 768699,
                                  "photos": Array [],
                                }
                              }
                              id={768699}
                              key="768699"
                            >
                              <div>
                                <Answer__Body>
                                  <div
                                    className="Answer__Body-sc-3mvxjy-6 gnmbP"
                                  >
                                    Laborum volupt aut et ex occaecati magni quas.
                                  </div>
                                </Answer__Body>
                                <div>
                                  <Photos
                                    photos={Array []}
                                  >
                                    <div />
                                  </Photos>
                                </div>
                                <Answer__BottomContainer>
                                  <div
                                    className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                                  >
                                    <Answer__NameDate>
                                      <div
                                        className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                      >
                                        By Ko2 on  July, 28 2020
                                      </div>
                                    </Answer__NameDate>
                                    |
                                    <Answer__Helpful>
                                      <div
                                        className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                      >
                                        Helpful?
                                      </div>
                                    </Answer__Helpful>
                                    <Answer__Yes
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                        onClick={[Function]}
                                      >
                                        Yes
                                      </div>
                                    </Answer__Yes>
                                    <Answer__Help>
                                      <div
                                        className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                      >
                                        (15)
                                      </div>
                                    </Answer__Help>
                                    |
                                    <Answer__Report
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                        onClick={[Function]}
                                      >
                                        Report
                                      </div>
                                    </Answer__Report>
                                  </div>
                                </Answer__BottomContainer>
                              </div>
                            </Answer>
                          </div>
                        </AnswerList>
                        <MoreA
                          display={false}
                          func={[Function]}
                        >
                          <div />
                        </MoreA>
                      </div>
                    </Item__ListContainer>
                  </div>
                </Item__BottomWrapper>
              </div>
            </Item>
          </div>
        </QuestionList__QList>
      </QuestionList>
      <QnA__BottomWrap>
        <div
          className="QnA__BottomWrap-sc-1hcdygh-2 dsWJWP"
        >
          <MoreQ
            buttonDisplay={false}
            func={[Function]}
          >
            <MoreQ__MQC>
              <div
                className="MoreQ__MQC-ir99cd-1 fxtzCf"
              />
            </MoreQ__MQC>
          </MoreQ>
          <AddQ
            func={[Function]}
            name="This is a product"
          >
            <AddQ__AQC>
              <div
                className="AddQ__AQC-fu61ov-1 bKyMVa"
              >
                <AddQ__AddAQ
                  onClick={[Function]}
                  type="submit"
                >
                  <button
                    className="AddQ__AddAQ-fu61ov-0 fsXLKI"
                    onClick={[Function]}
                    type="submit"
                  >
                    ADD A QUESTION +
                  </button>
                </AddQ__AddAQ>
              </div>
            </AddQ__AQC>
          </AddQ>
        </div>
      </QnA__BottomWrap>
    </div>
  </QnA__QnAContainer>
</QnA>
`;
